### PR TITLE
Igor/stats query

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: dist/znk.infra.js [hunk 1]
```diff
@@ -2667,81 +2671,119 @@
...
-                    });
+                if (!angular.isArray(optionalIds)) {
+                    optionalIds = [];
+                }
 
-                    if(currWeakestCategory.weakness === -Infinity){
-                        return null;
-                    }
+                return StatsSrv.getLevelStats(level).then(function (levelStats) {
+                    var iteratedObjProm = $q.when();
```
Comment (assafshp) [inline dist/znk.infra.js]: why do you need to resolve it immediately ?

File: dist/znk.infra.js [hunk 2]
```diff
@@ -2667,81 +2671,119 @@
...
+                                    iteratedObj[categoryKey] = new StatsSrv.BaseStats(categoryId, true);
+                                    iteratedObj[categoryKey].parentsIds = parentsIds;
+                                });
+                                allProm.push(prom);
+                            }
+                        });
+                        iteratedObjProm = $q.all(allProm);
+                    } else {
+                        iteratedObjProm = $q.when();
+                        iteratedObj = levelStats;
```
Comment (assafshp) [inline dist/znk.infra.js]: why not to return it in the resolve (line above) ?

File: dist/znk.infra.js [hunk 3]
```diff
@@ -2771,9 +2813,11 @@
 
                 var StatsSrv = {};
 
-                function _getCategoryLookup() {
-                    return $injector.invoke(getCategoryLookup);
-                }
+                var _getCategoryLookup = function() {
+                    return $injector.invoke(getCategoryLookup).then(function(categoryMap){
+                        return categoryMap;
```
Comment (assafshp) [inline dist/znk.infra.js]: how did it work before without the return ?

File: dist/znk.infra.js (hunk 1 above)
Comment (igormas) [inline dist/znk.infra.js]: In case no optional ids were provided then i have no async operations.

File: dist/znk.infra.js (hunk 2 above)
Comment (igormas) [inline dist/znk.infra.js]: because the iteratedObj can be provided straight from the $q.all function.

File: dist/znk.infra.js (hunk 3 above)
Comment (igormas) [inline dist/znk.infra.js]: the return wasn't there before, the return was straight from the injector invoke.

**Pull request merged**